chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -243,27 +243,37 @@ out:
 					"subscription: txid=%v, numconfs=%v",
 					msg.TxID, msg.NumConfirmations)
 
-				_, currentHeight, err := b.chainConn.GetBestBlock()
-				if err != nil {
-					chainntnfs.Log.Error(err)
-				}
+				currentHeight := uint32(bestHeight)
+
+				// Look up whether the transaction is already
+				// included in the active chain. We'll do this
+				// in a goroutine to prevent blocking
+				// potentially long rescans.
+				go func() {
```

Comment:
Not sure if it's really needed here 🤔

Reply:
Yeah I don't think it's needed here. Otherwise a very distant rescan can cause the shutdown process to be held up.

Reply:
we could thread a quit chan down to the rescan if we want. thinking it might be a good idea actually, to prevent the rescan from causing an accidental panic after the chain conns are deinitialized

Reply:
I think it's a nice pattern to follow, as also test failures tend to be harder to trace if there are stray go routines remaining.

Reply:
Fixed.